WordPress/Classes/ViewRelated/Suggestions/SuggestionsTableView.swift

```diff
@@ -9,7 +10,126 @@ extension SuggestionType {
     }
 }
 
-@objc extension SuggestionsTableView {
+@objc extension SuggestionsTableView: UITableViewDataSource, UITableViewDelegate {
+
+    // MARK: - Static Properties
+
+    private static let nonEmptyString = "_"
+
+    // MARK: - UITableViewDataSource & UITableViewDelegate
+
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        return viewModel.isLoading ? 1 : viewModel.sections.count
+    }
+
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.isLoading ? 1 : viewModel.sections[section].rows.count
+    }
+
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "SuggestionsTableViewCell", for: indexPath) as! SuggestionsTableViewCell
+        if viewModel.isLoading {
+            cell.titleLabel.text = NSLocalizedString("Loading...", comment: "Suggestions loading message")
+            cell.subtitleLabel.text = nil
+            cell.iconImageView.image = nil
+            cell.selectionStyle = .none
+            return cell
+        }
+
+        cell.selectionStyle = .default
+
+        let suggestion = viewModel.sections[indexPath.section].rows[indexPath.row]
+        cell.titleLabel.text = suggestion.title
+        cell.subtitleLabel.text = suggestion.subtitle
+        self.loadImage(for: suggestion, in: cell, at: indexPath, with: viewModel)
+
+        return cell
+    }
+
+    public func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        // Ideally, we should be returning `nil` instead of `Self.nonEmptyString`.
+        // But doing that doesn't hide the table section headers and footers.
```

Comment:
Would we be able to utilize `tableView(_:heightForHeaderInSection:)` (and footer) to hide the [headers](https://developer.apple.com/documentation/uikit/uitableviewdelegate/1614855-tableview) and [footers](https://developer.apple.com/documentation/uikit/uitableviewdelegate/1614967-tableview)? (e.g. return a 0 height in this case)

Reply:
Those height methods get called only when `tableView(_:titleForHeaderInSection:)` and `tableView(_:titleForFooterInSection:)` return a **non-nil and non-empty string**. If they're not called, the section header / footer will be visible.

So to hide the section header, all we have to do is to implement `tableView(_:heightForHeaderInSection:)` like this and it should work:

```
public func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
    return viewModel.sections.count > 1 ? tableView.sectionHeaderHeight : .leastNonzeroMagnitude
}
```

But to hide the section footer, we have to implement `tableView(_:titleForFooterInSection:)`even though we are not planning to display any text.

```
public func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
   return viewModel.sections.count > 1 ? nil : Self.nonEmptyString
}

public func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
   return viewModel.sections.count > 1 ? tableView.sectionFooterHeight : .leastNonzeroMagnitude
}
```

I think I prefer this approach because it keeps the section header / footer hiding logic in one place and close to it's to each other. Pushed ✅

Let me know if something is not clear. 🙏